rpmlint/checks/SUIDPermissionsCheck.py

```diff
@@ -0,0 +1,212 @@
+                    else:
+                        f += '/'
+
+                entry = self.perms[f]
+
+                if stat.S_ISREG(mode) and mode & (
+                    stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH
+                ):
+                    # pie binaries have 'shared object' here
+                    if (
+                        pkgfile.magic.startswith('ELF ') and ('shared object' not in pkgfile.magic) and ('pie executable' not in pkgfile.magic)
+                    ):
+                        self.output.add_info(
+                            'E', pkg, 'non-position-independent-executable', f
+                        )
+
+                m = entry.mode
+                o = ':'.join((entry.owner, entry.group))
+
+                if stat.S_IMODE(mode) != m:
+                    self.output.add_info(
+                        'E',
+                        pkg,
+                        'permissions-incorrect',
+                        '%(file)s has mode 0%(mode)o but should be 0%(m)o'
+                        % {'file': f, 'mode': stat.S_IMODE(mode), 'm': m},
+                    )
+
+                if owner != o:
+                    self.output.add_info(
+                        'E',
+                        pkg,
+                        'permissions-incorrect-owner',
+                        '%(file)s belongs to %(owner)s but should be %(o)s'
+                        % {'file': f, 'owner': owner, 'o': o},
+                    )
+
+            elif not stat.S_ISLNK(mode):
+
+                if f + '/' in self.perms:
+                    self.output.add_info(
+                        'W',
+                        pkg,
+                        'permissions-file-as-dir',
+                        f + ' is a file but listed as directory',
+                    )
+
+                if mode & (stat.S_ISUID | stat.S_ISGID):
+                    need_verifyscript = True
+                    msg = (
+                        '%(file)s is packaged with '
+                        'setuid/setgid bits (0%(mode)o)'
+                        % {'file': f, 'mode': stat.S_IMODE(mode)}
+                    )
+                    if not stat.S_ISDIR(mode):
+                        self.output.add_info(
+                            'E', pkg, 'permissions-file-setuid-bit', msg
+                        )
+                    else:
+                        self.output.add_info(
+                            'W', pkg, 'permissions-directory-setuid-bit', msg
+                        )
+
+                    if stat.S_ISREG(mode):
+                        if (
+                            'shared object' not in pkgfile.magic and 'pie executable' not in pkgfile.magic
+                        ):
+                            self.output.add_info(
+                                'E', pkg, 'non-position-independent-executable', f
+                            )
+
+            script = pkg[rpm.RPMTAG_POSTIN] or pkg.scriptprog(rpm.RPMTAG_POSTINPROG)
```

Comment:
The rest of the `for` loop body that deals with verification of the %post scriptles should also go into a separate member function for better readability.